benchmark kernel header downloading

usamasaqib opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Draft PRs should be prefixed with `[WIP]` in their title.

-->
### What does this PR do?
This PR adds benchmarking for kernel header downloading for debian and ubuntu.

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succint form, consider
  opening multiple pull requests instead of a single one.
-->

### Motivation

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

hush-hush committed:
Adding host metadata to the inventory payload

usamasaqib committed:
benchmark added for ubuntu header downloading

usamasaqib committed:
benchmark added for ubuntu header downloading

usamasaqib committed:
bench marking for debian header download added

usamasaqib committed:
bench marking for debian header download added

usamasaqib committed:
facility added to pass arbitrary flags to 'go test'

usamasaqib committed:
facility added to pass arbitrary flags to 'go test'

albertvaka committed:
gofmt pre-commit-hook: update the file and simplify by default (#12298)

This was the behavior before #11750

djmitche committed:
Add comments about secrets to flare code (#12285)

usamasaqib committed:
linter errors fixed

usamasaqib committed:
linter errors fixed

Commit:
Updating Google Marketplace dependency for the Datadog Operator (#12286)

hush-hush committed:
Remove 'CreatePayload' from the public API of inventories

hush-hush committed:
Support 'enable_metadata_collection' configuration for inventories metadata

brycekahle committed:
Do not commit changes to runtime integrity hashes to repo (#12261)

* Remove hashes from runtime built files

* Don't commit hash changes to repository

gh123man committed:
Journald tailer support for tailing modes (#12276)

This PR adds support for the logs config tailing modes: beginning, end, forceBeginning, forceEnd. This resolves #11072

agocs committed:
[Serverless] Always call TellDaemonRuntimeStarted before TellDaemonRuntimeDone (#12287)

* synchronously call TellDaemonRuntimeStarted before running invocationHandler asynch

* Add a test to assert that HandleInvocation won't accidentally segfault

* defer that recover

* Revert "synchronously call TellDaemonRuntimeStarted before running invocationHandler asynch"

This reverts commit 8682842e9202a4984a38b00fdf427837c9e2d46b.

* Revert "Revert "synchronously call TellDaemonRuntimeStarted before running invocationHandler asynch""

This reverts commit 5bc459340be2058fa1b929f60a07b539db5839a6.

* Run this test in a loop

* the word is SIGSEGV

* trigger CI

nplanel committed:
[network] fix openssl when no fd are used (#12078)

We capturing the connection tuple when SSL_do_handshake is called.

typical callgraph
SSL_do_handshake(ssl_ctx)
    maps/ssl_ctx_by_pid_tgid[pid] = ssl_ctx 
     write(socket)
           tcp_sendmsg(conn_tuple)
              ssl_ctx = maps/ssl_ctx_by_pid_tgid[pid]
              maps/ssl_sock_by_ctx[ssl_ctx] = conn_tuple
    delete(maps/ssl_ctx_by_pid_tgid, pid)

Motivation
This would add support on haproxy sever side as openssl is used via BIO memory buffer
Fix >=5.5.0 kernel socket/connect/read/write only calls #10656

Co-authored-by: Pedro Lambert <pedrolambert@gmail.com>

amenasria committed:
[slack-notifier] Truncate very long job names (#12306)

Truncate long job names and set the max_char_per_job limit to 48

keisku committed:
Support OTLP ExponentialHistogram (#12083)

mx-psi committed:
[pkg/otlp] Remove usages of `config.Map.Set` (#12296)

KSerrania committed:
[notifications] Separate infrastructure failures from regular job failures (#12324)

Adds a new section to the pipeline Slack notifications to separate infrastructure failures from regular job failures.
Only adds these infrastructure failures to the main Slack message, not the messages sent to teams.
Jobs are discriminated based on job logs containing specific indications that there was an infrastructure failure.

L3n41c committed:
Fix Windows container startup and `agent status` hang (#12275)

djmitche committed:
Use go.uber.org/atomic instead of sync/atomic in systray (#12319)

ahmed-mez committed:
[CI][revive] Add comments for public variables/functions (#12326)

juliogreff committed:
Add telemetry about container event bundling

The docker and kubernetesapiserver checks bundle events coming from the
Docker daemon and API Server, respectively, based on a few rules. We
want to monitor how that bundling works, and estimate the impact of a
possible unbundling of those events, so we are adding telemetry to that
effect.

This commit adds the following metrics:

* kubernetes_apiserver_kube_events: kubernetes events processed by the
  check
* kubernetes_apiserver_emitted_events: number of bundles emitted as
  datadog events
* docker_events: docker events processed by the check
* docker_emitted_events: number of bundles emitted as datadog events

ogaca-dd committed:
Enable govet (#12273)

* Enable govet

brycekahle committed:
Revert "Do not commit changes to runtime integrity hashes to repo (#12261)" (#12332)

This reverts commit 307d4bd4064ec797f93c4f0c3444efb5232af5c5.

hush-hush committed:
Fix inventory warnings not being printed

KSerrania committed:
[notifications] Add more infra failure filters (#12329)

Follow-up of #12324, adds one more log filter to check for infrastructure failures.

mbotarro committed:
fix unreachable code (#12336)

brycekahle committed:
Add more accurate UDP recv path in runtime tracer (#12279)

* Add more accurate UDP recv path in runtime tracer

* python lint

* python lint

* Add __maybe_unused

* python lint

* init head pointer

* memset iphdr, ipv6hdr, udphdr, and tcphdr

* Init net_head

* Init trans_head

brycekahle commented on pkg/util/kernel/download_headers_test.go:
```diff
@@ -0,0 +1,213 @@
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		// create output directory for kernel headers
+		if err := os.MkdirAll(kname, 0744); err != nil {
```
I think we should try to use `t.TempDir()` if we can: https://pkg.go.dev/testing@go1.18.3#T.TempDir

safchain committed:
[CWS] fix mutable variable test (#12333)

* refactor error handling in secl rule compilation

* make yaml not convert value to boolean

FlorianVeaux committed:
Refactor gosnmplib module to pkg/snmp (#12098)

* Refactor gosnmplib module to pkg/snmp

* Address review

paulcacheux committed:
[CWS] start and close policy providers (#12310)

paulcacheux committed:
[CWS] fix CWS e2e test - add support for local arm64 runs (#12325)

* [CWS] e2e: fix `app`/`App` confusion

* [e2e] implement local e2e for arm64 based dev envs

* [e2e] use pined version of `alpine/helm` docker image

usamasaqib committed:
Using the TempDir function to create temporary directories, when appropriate

usamasaqib committed:
Using the TempDir function to create temporary directories, when appropriate

djmitche committed:
Move some types and values out of pkg/logs/config (#12248)

* move LogSource and LogSources to pkg/logs/sources

* move LogStatus to pkg/logs/internal/status

Note that this leaves pkg/logs/status alone.  That is a different kind
of status, is global, and is tied in with some host-metadata stuff
outside of pkg/logs, so it can't move to pkg/logs/internal.  See
AML-190.

* move Info types into pkg/logs/internal/status

hush-hush committed:
Adding documentation about hostname detection

The current hostname detection is complicated. This commit documents its
logic so it's simpler to rework it later.

djmitche committed:
cca_in_ad -> create cca templates and filter them (#12320)

With this change, when logs_config.cca_in_ad is set, AD config providers
for containers will add a `container_collect_all` configuration template
for each container.

It then falls to the config reconciliation process to avoid
double-logging containers that have more-specific configuration.  This
is accomplished with a per-service template filter which drops the
container_collect_all configuration iff there is another template with a
nonempty LogsConfig.

djmitche committed:
cca_in_ad -> do not send service configs (#12321)

AD has historically sent "service configs" as a way to inform the logs
agent about containers, so that it can implement container_collect_all.
This is no longer necessary when CCA is handled in AD.

scottopell committed:
Re-enable SA2001 checks (#12295)

* pkg/logs: disable SA2001 check for lock-unlock pattern

* golangci: Enables SA2001 checks

sschepens committed:
[prometheus] fix check metrics parsing (#12318)

* skip prometheus checks metrics parsing to avoid breaking multiple possible formats

* use interface{} instead of json raw message, add tests

* fix tests

* add release notes

* Update releasenotes/notes/fix-prometheus-checks-metrics-07509580eb216c76.yaml

Co-authored-by: Jorie Helwig <joriephoto@gmail.com>

Co-authored-by: Jorie Helwig <joriephoto@gmail.com>

Barbayar committed:
pkg/trace/sampler: Fixes Rare Sampler logic (#12341)

brycekahle committed:
Regenerate sample docker manifests (#12111)

mbotarro committed:
[process-agent] Add the Event Store (#12159)

* implement the Event Store

maxday committed:
fix bootstrap script name in integration tests (#12348)

gh123man committed:
fix tcp launcher/tailer deadlock on agent shutdown (#12316)

* fix tcp launcher/tailer deadlock

* reno

* Update releasenotes/notes/fix-tcp-listener-deadlock-d441552a6e64263e.yaml

Co-authored-by: Kaylyn <kaylyn.sigler@datadoghq.com>

* Update releasenotes/notes/fix-tcp-listener-deadlock-d441552a6e64263e.yaml

Co-authored-by: maxime mouial <hush-hush@users.noreply.github.com>

Co-authored-by: Kaylyn <kaylyn.sigler@datadoghq.com>
Co-authored-by: maxime mouial <hush-hush@users.noreply.github.com>

p-lambert committed:
[networks] Use NAT information to generate `ByteKey` (#12288)

* Add log entry for closed connections

* Use NAT addresses for `ByteKey`

* [networks] Add test

* [experimental] Add multi-approach to `ByteKey` generation

* Add `closedKey` type

* Add test for address collision

* Add `ByteKeyNAT`

* Remove `closedKey` type

davidor committed:
[workloadmeta] Return empty in `ListContainers` instead of err (#12355)

amenasria committed:
[CI] Fix `revive` linter agent-core errors (#12356)

Add comments to fix revive linter

Co-authored-by: Nicolas Guerguadj <nicolas.guerguadj@epita.fr>

paulcacheux committed:
[CWS] fix cws e2e tests (#12351)

* [CWS] skip reloading do not reload policies after init self tester

* [CWS] rework watch mechanism from PoliciesDirProvider

* [CWS] add missing `BindEnvAndSetDefault` for watch dir option

* [CWS] sync tries/delay between metric and logs

ahmed-mez committed:
[KSM Core] Add ingress metrics (#12344)

amenasria committed:
Add comment to fix revive linter (#12357)

paulcacheux committed:
Regenerate sample docker manifests (#12360)

hush-hush committed:
Mock config now register a cleanup function

The mock config now registers a cleanup function to the current test to
reset `config.Datadog` to it's original state. This prevents tests that
use the mock to leave their configuration state to following tests not
using the Mock.

mx-psi committed:
Bump OpenTelemetry dependencies to v0.53.0 (#12343)

* Bump Collector dependencies to v0.52.0

* Bump Collector dependencies to v0.53.0

* Update licenses

* Add release note

* Update releasenotes/notes/otlp-v0.53.0-bf8295972049ef32.yaml

Co-authored-by: Jorie Helwig <joriephoto@gmail.com>

Co-authored-by: Jorie Helwig <joriephoto@gmail.com>

Commit:
[tasks] Remove the --redo flag of release.build-rc (#12342)

djmitche committed:
Upgrade to Go1.17.11 (#12358)

amenasria committed:
[CWS] Add comment to fix revive linter (#12354)

amenasria committed:
[CI] Fix `revive` linter container-integrations errors (#12352)

Add comments to fix revive linter

amenasria committed:
[CI] Fix `revive` linter errors (#12353)

Add comments to fix revive linter

trishankatdatadog committed:
Update CODEOWNERS (#12221)

* Update CODEOWNERS

* Fix typo in proposed CODEOWNERS

scottopell committed:
fix: Trims any surrounding whitespace from auth token as its read from disk (#12359)

paulcacheux committed:
fix revive missing comments (#12368)

davidor committed:
[workloadmeta] Add ListContainersWithFilter func (#12213)

* [workloadmeta] Add ListContainersWithFilter

* Use new Store#ListContainersWithFilter

* [compliance/agent/telemetry_test] Check non-running container

djmitche committed:
Allow subscribing to all LogSources (#12339)

brycekahle committed:
Isolate network namespaced tests from each other (#12377)

This should help some spurious errors I've seen locally.

scottopell committed:
Redact any Bearer tokens present in 'agent status' output (#12338)

* scrubber: Adds 'Bearer' scrubbing to default set of scrubbers

* agent-status: Scrubs errors before printing and uses logger instead of raw prints

* Adds release notes about this change

* Linting errors

* fixes test relying on 'status' output being plain json

* Adds missing require

* unlink agent-status json tempfile explicitly

IvanTopolcic committed:
[Serverless] SLS-2094: Get KMS/Secrets Managers values from the Extension (#12155)

Add support for decrypting secrets within the Datadog Lambda Extension

vboulineau committed:
Add a self cgroup reader (#12220)

nplanel committed:
[go deps] update package release as go.sum/sum.golang.org mismatch (#12305)

vboulineau committed:
Fix TLS Handshakes with leader forwarding, decrease loglevel to warning (#12373)

paulcacheux committed:
[CWS] fix listeners race (#12378)

* [CWS] add cmd flag to run functional tests with race detector

* [CWS] add `listenersLock` to protect the `listeners` field in `RuleSet`

* [CWS] improve race flag

ameske committed:
[RCM] Add a method to pull only the TUF version state (#12370)

* [RCM] Add a method to pull only the TUF version state

Pulling the full state is a non-trivial operation, and we were doing it
on every update request from an agent/tracer client. These clients ask
for updates quite frequently and this was causing the agent to use
more CPU than intended for remote configuration updates.

To fix this, we split out full state checks (which ARE needed for the
agent's CLI for users to query the remote config state) from TUF Version
state checks, which is all that is needed for trace/agent client updates
as well as getting updates from the remote configuration backend.

* Use constants for TUF metadata

* Consolidate to just TUFVersionState

* Adapt tests to account for State vs TUFVersionState

* update labels for PR

AlexandreYang committed:
[netflow] Add NDM NetFlow feature (#10976)

Co-authored-by: pducolin <45568537+pducolin@users.noreply.github.com>

maxday committed:
[Serverless] JSONify integration tests  (#12340)

kangyili committed:
move process chunking code to util package (#12323)

lebauce committed:
[CSPM] Downgrade rego to 0.37.2 (#12387)

* Downgrade opa to v0.37.2

* Revert rego changes requiring opa > 0.37.2

AlexandreYang committed:
[netflow] Add NDM NetFlow static enrichments (#12049)

safchain committed:
fix regexp parsing (#12380)

paulcacheux committed:
[CWS] respect `EnableNetwork` when configuring functional tests module (#12256)

paulcacheux committed:
[CSPM] improve rego done status reports (#12330)

* [CSPM] improve rego done status reports

* [CSPM] improve error report debug log

dependabot[bot] committed:
Bump golang.org/x/tools from 0.1.10 to 0.1.11 in /pkg/security/secl (#12392)

* Bump golang.org/x/tools from 0.1.10 to 0.1.11 in /pkg/security/secl

Bumps [golang.org/x/tools](https://github.com/golang/tools) from 0.1.10 to 0.1.11.
- [Release notes](https://github.com/golang/tools/releases)
- [Commits](https://github.com/golang/tools/compare/v0.1.10...v0.1.11)

---
updated-dependencies:
- dependency-name: golang.org/x/tools
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* Auto-generate go.sum and LICENSE-3rdparty.csv changes

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: dependabot[bot] <dependabot[bot]@users.noreply.github.com>

paulcacheux committed:
[CWS] only reset rename state if syscall is supported (#12398)

safchain committed:
add structured field definition (#12396)

julien-lebot committed:
Fix [Windows][Installer] Don't consider NT AUTHORITY as a domain. (#11643)

julien-lebot committed:
[Windows][Installer] Support gMSA accounts (#12372)

keisku committed:
pkg/otlp: Add `instrumentation_scope_metadata_as_tags` (#12393)

* add instrumentation_scope_metadata_as_tags, deprecate instrumentation_library_metadata_as_tags

* reno new

* return error if both configs are enabled

* fix comments

YoannGh committed:
[CWS] Add new process.is_thread field (#12303)

* add a field to test whether a process is considered a thread

* update documentation for new is_thread field

* add TestProcessIsThread functional tests

* update documentation to reflect code refactoring

* add release note

* Review: fix documentation and release note

* refactor code to expose is_thread to every process context

* add is_thread to process tree and avoid using a resolve function

Co-authored-by: Ursula Chen <58821586+urseberry@users.noreply.github.com>

djmitche committed:
Move file tailer's openFile to pkg/util/filesystem (#12379)

* Move file tailer's openFile to pkg/util/filesystem

This is a windows-only reimplementation of os.Open that allows
(some) concurrent use of the file.  It will be useful in checking
file access in the container launcher.

* update doc comment

paulcacheux committed:
Bump fake-datadog dependencies (#12407)

ogaca-dd committed:
Flush sketches (#12255)

* Extract the generic part of IterableSeries to iterableMetrics
* Remove unused SketchSeriesList.MarshalJSON method
When `useV1API` is false `serializePayload` calls only `serializePayloadProto`
* Update `SketchSeriesList` from  `[]SketchSeries` to `[]*SketchSeries`
- Be consistent with the type `Series`
- Avoid extra copies when `IterableSketches` willl be introduced
* Introduce SketchesSink
* Move iterableMetrics to its own file.
No code changed.
* updateHugeSketchesTelemetry takes `*metrics.SketchSeries` instead of `*metrics.SketchSeriesList`
* Introduce SketchesSource
* Add IterableSketches
* Replace SketchSeriesList by SketchesSourceTest in unit tests
SketchesSourceTest will not implement SketchesSource
* Add a method to IterableSketches to know if there is something to serialize
* Replace StartIteration by StartSerialization
* Fix a compilation issue in otlp after rebasing code from main
* Rename StartSerialization to Serialize
* Add a comment for `// Don't send empty sketches payloads`
* Use sketchesErr in consumer.go

brycekahle committed:
Only run resolver test that requires root if root (#12399)

djmitche committed:
cca_in_ad -> disable CCA scheduler (#12376)

When CCA is being handled by AD, there's no need for a CCA scheduler in
the logs agent.

djmitche committed:
cca_in_ad -> do not schedule logs services (#12375)

When `logs_config.cca_in_ad` is set, the logs-agent no longer concerns
itself with "services" -- only with sources.

djmitche committed:
Add some minor docs to AD, Logs-Agent (#12410)

paulcacheux committed:
system-probe: ensure integrity checks are in sync (#12114)

* Add `inv -e system-probe.generate-runtime-files` to `security_go_generate_check` CI job

* Fix wrong integrity check

* Re-gen integrity checks

* Re-gen integrity checks

julien-lebot committed:
Don't fail install if we cannot detect whether or not the user account is a managed service account (#12411)

julien-lebot committed:
Bump pyjwt version to 2.4.0 (#12403)

maxday committed:
[Serverless] filter out empty logs to prevent snapshot diff flakiness (#12413)

mx-psi committed:
[pkg/otlp/model] Partial sync with Datadog exporter codebase (#12346)

* [pkg/otlp/model] Partial sync with OpenTelemetry Collector codebase

* Add `UsePreviewHostnameLogic` field to configure the renaming.

paulcacheux committed:
[CSPM] bump security agent policies to v0.26.4 (#12419)

monganai committed:
Include `log_level` in flare file names (#12236)

Appends `log_level` to the flare archive name during creation flare creation.

mbotarro committed:
[process-agent] Add the Cleanup method to the Check interface (#12389)

extend the Check interface with a Cleanup method

juliogreff committed:
[workloadmeta] Expire entities instantly

Instead of waiting anywhere between 10 and 15 seconds to remove entites
that no longer appear in the lists returned by upstream (such as k8s
apiserver, podman, ECS APIs), we know that they're gone so there's no
need to keep them around anymore. It's quite possible that this delay
was present before so the tagger could cache tags for longer than the
actual workloads, but since the tagger now has its own expiry logic,
this is no longer needed.

albertvaka committed:
pkg/trace/api: add generic EVP Proxy API (#12271)

bwoebi committed:
pkg/obfuscate: fix parsing negative numbers being mistakenly interpreted as octal (#11129)

Any number matching -\d0\d*[89]\d* format was misrecognized as an octal
number due to the parser ignoring the first digit.

Additionally Any number -.\d+ was not interpreted as a single number, but
as distinct tokens '-' and a number. Added an additional case to handle
this scenario to ensure proper normalization.

paulcacheux committed:
[CSPM] fix expected findings, with host-selectors reintroduction (#12431)

Kaderinho committed:
use a mock clock in dogstatsd unit tests (#12422)

* tests(dogstastd): add a clock in DSD debug server and use a mock clock in unit tests

Signed-off-by: Nicolas Guerguadj <nicolas.guerguadj@epita.fr>

usamasaqib committed:
Namespace routers for modules by name. (#12148)

* NETAGENT-116: added map of router in loader. Module name indexes a namespaced router for the module

* NETAGENT-116: making a namepsaced subrouter for each module

* prefixing module names to all queries to module endpoints

* Converted endpoint URLs to constant strings

* '../check/..' added in path

* endpoint paths for windows prefixed by module name

* check endpoint for oom_kill_probe and tcp_queue_length_tracer prefixed by module name

* seperated datadog package imports from standard package imports

* package import order cleanup

usamasaqib committed:
[networks] Add amazon linux 2 for testing in CI (#12394)

* amazon linux 2 added to CI kitchen setup

* fixed kernel headers package name for amazon linux

* package name for netcat added

* tracer_test log level set to debug

* fixed package name for kernel headers

* updating repositories when platform amazon is detected

* adding kernel version to amazon kernel headers package

* changed back tracer_test logging level to 'warn'

* formatting fix

hmahmood committed:
Return empty stats maps when features are disabled (#12424)

djmitche committed:
fix lint in #12148 (#12438)

hmahmood committed:
Add invoke command for generating kitchen config for system-probe tests (#12428)

IvanTopolcic committed:
Add trigger types and refactor (#12137)

NouemanKHAL committed:
Fix TestCCCache_GetCFApplications flakiness (#12445)

L3n41c committed:
Revert "Update Agent container base image to Ubuntu 22.04 (#11745)" (#12433)

This reverts commit 26f7615c7f03df27477572c83f9f26cefa8dbc63.

coignetp committed:
[RCM] Send tuf error back (#12421)

* Send tuf error back to datadog

* Fix tests

albertvaka committed:
Make evp_proxy configurable via env vars (#12439)

Calls BindEnv for the new settings we added in #12271
Also renames files form evp_intake_proxy to evp_proxy.

AlexandreYang committed:
[netflow] Use default max events per payload (#12437)

Kaderinho committed:
Add more endpoints for connectivity diagnose (#12250)

* feat(diagnose): add v1 and v2 metrics endpoints used by the forwarder in the diagnose command

* refactor(diagnose): remove apiKeyInQueryString attribute as the API key is passed through HTTP headers

Signed-off-by: Nicolas Guerguadj <nicolas.guerguadj@epita.fr>

Co-authored-by: Dustin J. Mitchell <dustin.mitchell@datadoghq.com>

mfpierre committed:
Enable Ingresses collection by default (#12385)

paulcacheux committed:
[CWS] policy versioning (#12412)

* [CWS] implement agent version constraint in SECL rule definitions

* [CWS] add utility to parse agent version semver

* [CWS] pass agent version where needed

* fix bugs

* [CWS] add tests

* [CWS] fix `check policies` cli cmd

* [CWS] add test for loading error mechanism

paulcacheux committed:
[CWS] add missing doc comment on `DefaultPolicyName` (#12458)

paulcacheux committed:
[CWS] fix macro loading (#12307)

* [CWS] remove `UserCtx` from `eval.Opts`

* [CWS] split `rules.Opts` from `eval.Opts`

* [CWS] introduce `MacroStore` to store macros out of options

* [CWS] fix tests

* [CWS] `EvalReplacementContext` -> `ReplacementContext`

* [CWS] fix more tests

* [CWS] fix cmd build

* [CWS] include replCtx in eval State

Gui774ume committed:
[CWS] Add dns events in activity dumps (#11466)

* add dns events in activity dumps

* merge DNS requests type

Kaderinho committed:
fix(tests, dogstatsd): add timeout after disabling debug loop to avoid data race (#12453)

Signed-off-by: Nicolas Guerguadj <nicolas.guerguadj@epita.fr>

L3n41c committed:
Revert "Fix `docker_build_cluster_agent_arm64` job (#12022)" (#12459)

This reverts commit 66ecb533d231f55ffafbb2cbc0a632a8798470fd.

keisku committed:
config_template: add `logs_config.open_files_limit` (#12450)

juliogreff committed:
[kubelet] Remove unused code

juliogreff committed:
[AD] Re-introduce autodiscovery_scheduled_configs telemetry

This was reverted in #10499 due to a bug in AD scheduling. This has
since been fixed in #11639, so I'm re-introducing equivalent changes as
to what #10499 reverted.

juliogreff committed:
[AD] Sort tags on resolved config's integration.Data

This fixes a [flaky
test](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/145953755#L1060)
due to unstable sorting order of hash keys in Go. It probably also makes
this resolution more stable if it ever has to merge in different tags at
runtime, by producing the same digest [1].

[1] That is, if the digest itself didn't have to handle tag ordering in
the first place (see pkg/autodiscovery/integration/config.go:366 at the
time of this writing). That can probably go away now, for a nice
performance improvement.

L3n41c committed:
Update some GO dependencies among which `opencontainers/runc` and `containerd/containerd` (#12455)

Kaderinho committed:
fix(tests, dogstatsd): increase the timeout after disabling debug loop because windows unit tests CI is broken (#12469)

KrisKennawayDD committed:
[Telemetry] Add missing telemetry metrics for HTTP log bytes sent. (#12417)

Populate the message.UnencodedSize for TCP logs and use it for the bytes_sent metrics.  This should be a functional NOP but helps to clarify the intended metric behaviour.

Gui774ume committed:
[CWS] Activity dump storage manager (#12084)

* introduce storage manager interface

* add remote storage and remote storage forwarder

* retry tagger multiple times

* update JSON format

* add activity dump encoding comment and reset default config

* fix CI

* change process monitoring serialization

Gui774ume committed:
fix uptime drift (#12475)

paulcacheux committed:
[CWS] reduce stack size in activity dump path (#12374)

* [CWS] reduce stack size in activity dump path

* [CWS] memset ad buffer to 0 before usage

paulcacheux committed:
CSPM: do not run docker rules on kubernetes environment (#12257)

* CSPM: do not run docker rules on kubernetes environment

* CSPM: fix tests

* CSPM: allow configuration of `skipOnKubernetes` for docker rules

* CSPM: fix tests

vboulineau committed:
Fix panic when container stats are missing (#12395)

YoannGh committed:
[CWS] Add new exit secl rule (#12116)

* add process exit code to exit event

* add new secl exit event

* do not inherit parent ExecTime to allow secl rules to match a process that called exec

* set release of exit event

* add function tests for exit secl rules

* initialize ExitEventSerializer

* update docs and code format

* ci: fix unnecessary conversion

* do not use process exec time to figure out if it has exec yet

* add release note

* Review: remove msg tag and fix typo

* remove unused function

* Review: update documentation

Kaderinho committed:
fix(tests): don't use assert.Eventually() in dsd debug loop unit tests as it still breaks the CI (#12476)

davidor committed:
[corechecks/ksm] Add tags for common k8s labels (#12479)

safchain committed:
release the raw buffer from the release function only (#12490)

paulcacheux committed:
Bump `fake-datadog` and add support for multiarch push (#12477)

* Add multiarch push step

* Bump `datadog/fake-datadog` tag to most recent

mx-psi committed:
[pkg/otlp] Backport changes from Datadog exporter (#12452)

* [pkg/otlp/model] Backport changes from Datadog exporter

* Adapt serializer and trace OTLP code to new changes

* Address comments on Kubernetes code

* [pkg/otlp/model] Rename to unknown source

* [pkg/trace] Address comments

* update ReceiveResourceSpans comment

Gui774ume committed:
use omitempty to differentiate activity dump and process monitoring serializations (#12491)

paulcacheux committed:
[CWS] improvements to jsonschema documentation extraction (#12102)

* Use `github.com/invopop/jsonschema` instead of `github.com/alecthomas/jsonschema`

* Re-gen documentation

* Extract doc from comments

* Doc conversion

* `inv -e generate-licenses`

* Generate documentation

* Generate documentation

* Re-gen documentation

* Apply doc review suggestions

* regen doc

AlexandreYang committed:
[netflow] Rename exporter to device in payload (#12480)

mbotarro committed:
[process-agent] Create the process_events check (#12390)

create the process_events check

sgnn7 committed:
[ci] Disable `golang_deps_generate` main branch builds (#12493)

`golang_deps_generate` is not handling recent changes to `go.mod`
properly (causing a memory exhaustion) and because of that is
breaking the main branch so we will remove it until we have the
problem(s) with this tool resolved.

safchain committed:
make use of readinto (#12147)

mbotarro committed:
Revert "[process-agent] Create the process_events check (#12390)" (#12496)

This reverts commit b6f79fec798a0d724ba9de90eef60ed51e1016df.

ajgajg1134 committed:
pkg/config: env var DD_OTLP_CONFIG_TRACES_SPAN_NAME_REMAPPINGS is json not bool (#12133)

hmahmood committed:
Use new RecordGetter and RecordHandler for perf callbacks (#12489)

keisku committed:
pkg/otlp: add logging exporter (#12281)

* pkg/otlp: add logging exporter

* inv generate-licenses

* reno new

* don't put invalid value in the pipeline config

* import go.opentelemetry.io/collector/config

* not to use deprecated methods

* rename to otlp_config.debug.log_level

* fix DebugLogEnabled

* fix errcheck

* rename to logLevel

* rename to loglevel

lebauce committed:
Fix panic when closing remote tagger (#12087)

arbll committed:
(rcm-251) avoid locking tracer/agent queries to RC when we poll (#12492)

L3n41c committed:
Update Agent container base image to Ubuntu 22.04 (#12481)

* Revert "Revert "Update Agent container base image to Ubuntu 22.04 (#11745)" (#12433)"

This reverts commit 315e8d0c03cb3de8a707a22a66a0c13e8f80873a.

* Revert "Revert "Fix `docker_build_cluster_agent_arm64` job (#12022)" (#12459)"

This reverts commit 5d1b658f45df598a5295f0d111bf706aff7883af.

* Make the seccomp profile hack probe the new syscalls to assess if they really need to be intercepted.

arbll committed:
(rcm) fix type and change name of backend state (#12484)

* (rcm) fix type and change name of backend state

* fix tests

albertvaka committed:
pkg/trace/api: forward container ID header in evp_proxy (#12500)

mx-psi committed:
[pkg/otlp] Skip faulty test on macOS (#12502)

ogaca-dd committed:
Add a rate limiter that reduce DogStatsD traffic  (#10867)

* Add a rate limiter that reduce DogStatsD traffic when the memory usage exceed a threshold.

* Fix compilation

* Remove `madvdontneed=1`

* Remove the division in geometricRateLimiter.keep

* Make waitWhileHighLimit more aggressive to reclaim the memory.

* Use `cgroups.NewSelfReader` to get the correct cgroup.

* Use a memory ballast

gh123man committed:
add debug log lines (#12495)

djmitche committed:
Use go.uber.org/atomic in cmd/system-probe (#12423)

Fixes possible alignment issues with atomic access in `cmd/system-probe`
by switching to `go.uber.org/atomic`

gh123man committed:
Add auto multiline lines matched metric telemetry and to status page.  (#12482)

* add auto multiline telemetry

* added comment

* Update releasenotes/notes/auto_multi_line_detection-telemetry-57409b313aa40c2d.yaml

Co-authored-by: May Lee <mayl@alumni.cmu.edu>

* improve debuyg logging

Co-authored-by: May Lee <mayl@alumni.cmu.edu>

Kaderinho committed:
add diagnose datadog-connectivity output to the flare archive (#12456)

* feat(flare): add diagnose datadog-connectivity output to the flare archive

Signed-off-by: Nicolas Guerguadj <nicolas.guerguadj@epita.fr>

amenasria committed:
[CI] Fix code to restore the exported documentation linter (#12441)

Remove some revive errors and re-enable linter ✨

amenasria committed:
[CI] Restore the exported documentation linter (#12312)

* Restore default linters

safchain committed:
fix compilation error caused by int overflow on some platform (#12501)

mx-psi committed:
[tasks] Split off module tagging from `release.tag-version` (#12268)

* [tasks] Split off module tagging from `release.tag-version`

* Address linter changes

mx-psi committed:
Update OpenTelemetry Collector dependencies to v0.54.0 (#12498)

* Update OpenTelemetry Collector dependencies to v0.54.0

* Update licenses

* Update release note to reflect new version and protocol version

* Downgrade testify to v1.7.2

* Fix tests code

ogaca-dd committed:
Don't log an error when series or sketches are not enabled (#12494)

* Don't log an error when series or sketches are not enabled

* Move release note to the right folder

paulcacheux committed:
[CWS] fix revive TODOs (#12513)

* [CWS] fix revive TODOs

* [CWS] fix missing revive TODOs

* regen doc

sgnn7 committed:
[pkg/dogstatsd] Fix linter problem in `mem_based_rate_limiter.go` (#12515)

This file had a comment that did not match the expected format for our
linter rules.

safchain committed:
do not handle time resolution error twice (#12514)

ameske committed:
[RCM] Error if required fields are missing in ClientGetConfigRequest (#12446)

We were returning a 200 status code and an empty body if a required
field to request updates was missing. This is an invalid request and
should have an error code instead of 200.

sgnn7 committed:
[pkg/serializer] Fix linter problem in `serializer.go` (#12516)

This file had a comment that did not match the expected format for our
linter rules on public methods.

mbotarro committed:
[process-agent] Create the process_events check (after IoT build fix) (#12511)

create the process_events check

AlexandreYang committed:
[netflow] Move namespace to device.namespace in payload (#12519)

gbbr committed:
pkg/trace/api: fix OTLP fallback hostname bug (#12520)

When no source is detected, the source must respect the fallback
hostname.

Barbayar committed:
pkg/trace/sampler: update RareSampler default values and add config (#12311)

Rare Sampler samples too much data traces for data storage services. Data storage services have high cardinality resources. As result, Rare Sampler ends up annotating too many resources as rare.
With this change, the rare sampler should sample less traces when the resource cardinality is high.

New default values are:
|         **Configuration**         | **Old Default** | **New Default** | **Config Name**                       |
|:---------------------------------|---------------:|---------------:|---------------------------------------:|
| Cool down for non priority traces |           2 min |           5 min | `apm_config.rare_sampler.cooldown`    |
| TPS                               |               5 |               5 | `apm_config.rare_sampler.tps`         |
| Cardinality                       |            1000 |             200 | `apm_config.rare_sampler.cardinality` |

vickenty committed:
Don't expire contexts for counters we may want to add later (#12337)

Agent keeps reporting counters for a fixed period of time after the
client app has stopped sending them. Contexts are expired on a
different schedule. When context expiration time is low enough,
granularity of flushes becomes a hindrance, as counters sometimes are
sampled not frequently enough to keep their contexts alive.

This patch prevents active counter contexts from being expired,
regardless of their timestamp in the context resolver. Once the
reporting logic is done with the context, it can be expired using the
regular process though.

mbotarro committed:
[process-agent] Use a separate domain for event payloads (#12460)

send events to a separate endpoint

gbbr committed:
pkg/trace/api: enchance OTLP hostname test (#12521)

nplanel committed:
ebpf build on kernel 5.18 (#12483)

kernel 5.18 define TASK_COMM_LEN in linux/sched.h and would need to be included for prebuild offset-guess

Stderr:

In file included from /lib/modules/5.18.5-arch1-1/build/include/linux/percpu-rwsem.h:7:
In file included from /lib/modules/5.18.5-arch1-1/build/include/linux/rcuwait.h:6:
In file included from /lib/modules/5.18.5-arch1-1/build/include/linux/sched/signal.h:7:
/lib/modules/5.18.5-arch1-1/build/include/linux/sched.h:282:2: error: expected identifier
        TASK_COMM_LEN = 16,
        ^
./pkg/network/ebpf/c/prebuilt/offset-guess.h:7:23: note: expanded from macro 'TASK_COMM_LEN'

julien-lebot committed:
Clarify in dev docs that the Agent is composed of multiple binaries (#12506)

gbbr committed:
pkg/trace/api: ensure best effort setting container tags in OTLP ingest (#12523)

This change makes sure that whenever we are in a containerised
environment (AWS, ECS, Fargate) and we are missing a container ID, we at
least try to add the tag returned by the source provider, if available.

spikat committed:
Add bind events to activity dumps (#12240)

spikat committed:
Fix CI (#12526)

ISauve committed:
Update nikos dependency (#12517)

* Update nikos

* Update licenses

* Update licenses

ameske committed:
[RCM] Replace the remote config agent client w/ an RFC compliant client (#12391)

We had an existing remote config client for agent sub-processes, but in
the time since we first wrote that client, an RFC was published for
integrating with remote configuration in tracers and agent subprocesses.
That RFC introduced changes that made our existing client non-compliant.
This work transforms the client to match the RFC.

The client is split up across two packages. The "agent client" lives in
"pkg/config/remote" and consists of the networking layer that builds
requests and sends them over gRPC to the core agent. It leverages the
"state" package in "pkg/remoteconfig/stae" to process those updates and
store the state using the `Repository` type. This `Repository` type can
be utilized in the Go tracer client as well to prevent that work from
having to be implemented again.

The reason for the split is mostly that we're wanting to slowly migrate
remoteconfig code out of "pkg/config/remote" and into "pkg/remoteconfig"
for clarity. This lays the groundwork for doing that.

One important note is that the Go client technically goes further than
the RFC mandates with respect to TUF validation in order to make sure
that we can expand the RFC to support partial TUF validation down the
road.

Another important note is that the model for relaying updates to
interested product subsystems has changed. Previously we would create
channels to distribute updates, we now use a simple function callback
model. If an update is expensive to apply, the product subsystem is free
to setup a channel/goroutine pipeline and use their callback function to
schedule the work. This allows for more flexibility and keeps us from
forcing a product subsystem to use channels and goroutines.

djmitche committed:
Use go.uber.org/atomic in pkg/netflow (#12405)

Start using `go.uber.com/atomic` in `pkg/netflow` instead of built-in atomics
to ensure alignment of access on some platforms.

nplanel committed:
[network/ebpf] https enable arm64 on kernel >= 5.5.0 (#12485)

Enable https on arm64 
 * syscall openat2 need to be hooked as it's a new arch and older syscall open is not implemented
 * bpf_probe_read() would need _user() when copy data from the userspace the arch have different segment for userspace and kernelspace
 * bpf_probe_read_user() could return EFAULT if reading userspace pointer out of bound, so we fallback to fill_path_safe() if the first attempt fail.
 * SSL_connect() need to be hook to support libssl 1.0 (SSL_do_handshake() is not called)

paulcacheux committed:
Revert "[CWS] reduce stack size in activity dump path (#12374)" (#12532)

This reverts commit d860d455862ee081dc3c24e464c33545fb402dc8.

mbotarro committed:
set processes as code owners for process-agent status template (#12524)

ameske committed:
[RCM] Fix missing comments on exported fields (#12534)

Kaderinho committed:
add hints when there is a mismatch between http and https when the Agent communicate with an endpoint (#12512)

* feat(diagnose): add hints when there is a mismatch between http and https when the Agent communicate with an endpoint

Signed-off-by: Nicolas Guerguadj <nicolas.guerguadj@epita.fr>

mbotarro committed:
[process-agent] Add human format output for process events (#12471)

add human format output support for process events

Commit:
pkg/obfuscate: add support for PostgreSQL JSON operators (#12026)

Add support for PostgreSQL specific JSON and JSONB operators to the obfuscate package.  This commit adds special handling in the tokenizer for multi-character json operators (#>, #>>, #-, @>, <@, ?|, ?&, ->, ->>).  These operators will only be parsed when DBMSPostgres is passed as the DBMSType in the SQLConfig.  This fixes an issue where operators with a `#` were incorrectly interpreted as a comment.

ameske committed:
[RCM] Fix storage of backend state in client (#12530)

There was a bug in the communication between the backend and the agent
involving backend state that is supposed to be opaque to the agent and
its clients. (allowing the backend to store whatever it wants downstream
to help optimize the update process or recreate state if needed)

In order to fix the bug, we needed to introduce a new field and change
the data type from json.RawMessage to []byte.

mbotarro committed:
collect containerID with process events (#12473)

Kaderinho committed:
enable datadog-connectivity command (#12525)

* feat(diagnose): enable datadog-connectivity command

Signed-off-by: Nicolas Guerguadj <nicolas.guerguadj@epita.fr>

* Copy Nit

Co-authored-by: Austin Lai <76412946+alai97@users.noreply.github.com>

sgnn7 committed:
[pkg/metadata] Remove double resource payload collection during sending (#12061)

* [pkg/metadata] Remove double payload resolution during sending

We don't need to generate the payload twice in this block (once for
error check, one for sending) since it's the same data. Since this data
may not be cached, the impact may not be negligble either.

derekwbrown committed:
[kitchen tests] skip chinese NLA windows when running test-signed driver (#12497)

* DO NOT MERGE
Running test on last testsigned build to verify 2019cn

* Add special case code for not running on `cn` windows when using a
testsigned build.

* Revert release.json to release-signed ddnpm

* python formatter

lebauce committed:
[CWS] Use ring buffers instead of perf maps when available (#12156)

* Use ring buffers instead of perf maps when available

* Report status of some eBPF features

* Fix linters

* Only declare bpf_ringbuf_output on kernels >= 5.8

* Allow using specific version of kernel headers to compile eBPF

* Set default ring buffer size

* Fix reorderer closing

* Fix probe monitor wait group add

* Fix panic when reporting ring buffer lost events

* Fix size of ring buffer

* Add comments and build tags

* Define globals for perf ring buffers

* Regenerate protobuf files

* Add config knobs for events map type and size

* Bump system-probe and nikos build images

* Make revive linter happy

* Set max ring buffer size to 64 MB

* Update generated eBPF bundle files

* Remove useless space in .gitlab-ci.yml

* Use cilium/ebpf library to detect ring buffers support

* Address invoke script comment

* Add github.com/cilium/ebpf/features to LICENSE-3rdparty.csv

* Remove unnecessary conversion

* Set maximum page count for ring buffer

* Add missing comments

safchain committed:
add busybox support (#12522)

ofek committed:
Fix Omnibus builds to support v2 manifests (#12535)

* Fix Omnibus builds to support v2 manifests

* try

* ruby bad

carlosroman committed:
chore: Updated Python3 from 3.8.11 to 3.8.13 (#12507)

Gui774ume committed:
fix proc_cache key typo (#12536)

paulcacheux committed:
[CWS] bump security agent policies to v0.27.0 (#12544)

Gui774ume committed:
[CWS] Fix TC reload race (#12537)

* Fix TC probes reload and umount event race

* Remove enableNetwork

* update ebpf-manager

kacper-murzyn committed:
[release] Update release.json and Go modules for 6/7.38.0-rc.1 (#12543)

* Update release.json and Go modules for 6/7.38.0-rc.1

* Update release.json

L3n41c committed:
Fix `revert_latest_[67]` GitLab jobs (#12554)

hmahmood committed:
Fix: crash in perf callback (#12553)

nplanel committed:
[network/ebpf] __always_inline kretprobe do_sys_open / do_sys_openat2 (#12548)

kacper-murzyn committed:
7.37.0 changelog (#12533) (#12546)

* 7.37.0 changelog (#12533)

* 7.37.0 changelog

* Apply suggestions from code review

Co-authored-by: Srdjan Grubor <sgnn7@sgnn7.org>

Co-authored-by: Srdjan Grubor <sgnn7@sgnn7.org>

* Kacper murzyn/7.37.1 changelog (#12565)

* Update CHANGELOG for 7.37.1

* Manual update

Co-authored-by: Srdjan Grubor <sgnn7@sgnn7.org>

Gui774ume committed:
ignore invalid entries in /proc/modules (#12567)

katiehockman committed:
Revert "pkg/trace: use go.uber.com/atomic (#12120)" (#12568)

* Revert "pkg/trace: use go.uber.com/atomic (#12120)"

This reverts commit 2469fbac5d39d3cea002dbf119035b3d48142a17.

This is due serialization issues caused by this change.
Error message from `agent status`:
```
template: /trace-agent.tmpl:46:139: executing "/trace-agent.tmpl" at <.trace_writer.Bytes>: wrong type for value; expected float64; got map[string]interface {}
```

* Revert "pkg/trace: use go.uber.com/atomic (#12120)"

This reverts commit 2469fbac5d39d3cea002dbf119035b3d48142a17.

This is due to serialization issues resulting from PR #12120 which block the agent release.
Error message from `agent status` before revert:
```
template: /trace-agent.tmpl:46:139: executing "/trace-agent.tmpl" at <.trace_writer.Bytes>: wrong type for value; expected float64; got map[string]interface {}
```

Below is the revert steps to call out merge conflicts:
```
~ git revert 2469fbac5d39d3cea002dbf119035b3d48142a17
Auto-merging pkg/trace/agent/agent.go
Auto-merging pkg/trace/agent/agent_test.go
Auto-merging pkg/trace/api/responses.go
Auto-merging pkg/trace/go.mod
CONFLICT (content): Merge conflict in pkg/trace/go.mod
Auto-merging pkg/trace/sampler/dynamic_config.go
CONFLICT (content): Merge conflict in pkg/trace/sampler/dynamic_config.go
Auto-merging pkg/trace/sampler/prioritysampler_test.go
CONFLICT (content): Merge conflict in pkg/trace/sampler/prioritysampler_test.go
Auto-merging pkg/trace/sampler/rare_sampler.go
CONFLICT (content): Merge conflict in pkg/trace/sampler/rare_sampler.go
Auto-merging pkg/trace/sampler/remote_rates.go
CONFLICT (content): Merge conflict in pkg/trace/sampler/remote_rates.go
Auto-merging pkg/trace/sampler/remote_rates_test.go
CONFLICT (content): Merge conflict in pkg/trace/sampler/remote_rates_test.go
error: could not revert 2469fbac5... pkg/trace: use go.uber.com/atomic (#12120)
```

vboulineau committed:
Revert docker.mem.in_use to use RSS memory instead of total memory (#12563)

juliogreff committed:
KSM: use v1beta1 PodDisruptionBudget and CronJob resources

This fixes a regression introduced when we upgraded to KSM 2.4:
https://github.com/kubernetes/kube-state-metrics/pull/1491. This
restores metrics for CronJob and PodDisruptionBudget resources for
Kubernetes 1.19 and older.

AlexandreYang committed:
[netflow] Mark NetFlow config as known (#12572)

paulcacheux committed:
CWS/CSPM: bump security agent policies to v0.27.1 (#12577)

vboulineau committed:
Revert force usage of CRI v1 API (allows fallback to v1alpha2) (#12575)

BaptisteFoy committed:
fix(remote-config): Throw TUF errors when getting configs from Director (#12578)

* fix(remote-config): Throw TUF errors when getting configs from Director

Currently any TUF error happening when getting a config from the
Director results in a Not Found error. If our snapshot is expired or
simply does not exist, we want to be able to get a meaningful error.
If that change is not made, we may fall into a case where we don't know
what is happening when we have a TUF issue in the agent or the backend.

* Nudge pipeline

lebauce committed:
[CWS] Set default ring buffer size to a power of 2 and multiple of page size (#12569)

* Set default ring buffer size to a power of 2 and multiple of page size

* Use 64MB for less than 64 cpus and 128MB otherwise for ring buffer

kacper-murzyn committed:
Update release.json and Go modules for 6/7.38.0-rc.2 (#12590)

djmitche committed:
Create the atomic.Bool when creating a ProcessAgentCheck (#12596)

paulcacheux committed:
[CWS] remove prerelease and metadata before checking agent version (#12615)

safchain committed:
add process context support (#12582)

kacper-murzyn committed:
Update release.json and Go modules for 6/7.38.0-rc.3 (#12618)

Kaderinho committed:
use nil body for api key validation endpoint to make it work across all sites (#12629)

* fix(diagnose): use nil body for api key validation endpoint to make it work across all sites

Signed-off-by: Nicolas Guerguadj <nicolas.guerguadj@epita.fr>

usamasaqib committed:
Only adding correct repository for Ubuntu.

Aptly uses leveldb for storing remote repositories. We search for the
required pacakge in these repos. The order in which leveldb returns the
repo while iterating is not deterministic. This adds variability in the
benchmarks, based on how long it takes us to find the correct repo. To
prevent this, and stabilise the results, this commit only adds a single
repository to the Ubuntu backend for downloading kernel headers.

usamasaqib committed:
Only adding correct repository for Ubuntu.

Aptly uses leveldb for storing remote repositories. We search for the
required pacakge in these repos. The order in which leveldb returns the
repo while iterating is not deterministic. This adds variability in the
benchmarks, based on how long it takes us to find the correct repo. To
prevent this, and stabilise the results, this commit only adds a single
repository to the Ubuntu backend for downloading kernel headers.

brycekahle commented:
@usamasaqib I think you might need a rebase to fix CI.

usamasaqib committed:
Merge branch 'usama.saqib/benchmark-kernel-header-downloading' of github.com:DataDog/datadog-agent into usama.saqib/benchmark-kernel-header-downloading
